test-data/unit/check-protocols.test

```diff
@@ -2187,3 +2187,123 @@ else:
     reveal_type(cls)  # E: Revealed type is 'Type[__main__.E]'
 [builtins fixtures/isinstance.pyi]
 [out]
+
+[case testCallableImplementsProtocol]
+from typing import Protocol
+
+class Caller(Protocol):
+    def __call__(self, x: str, *args: int) -> None: ...
+
+def call(x: str, *args: int) -> None:
+    pass
+def bad(x: int, *args: str) -> None:
+    pass
+
+def func(caller: Caller) -> None:
+    pass
+
+func(call)
+func(bad)  # E: Argument 1 to "func" has incompatible type "Callable[[int, VarArg(str)], None]"; expected "Caller"
```

Comment:
Once we deprecate the old syntax, we should come up with a different syntax for "rich" callables in error messages as well.

Reply:
The format is already there, `MessageBuilder.pretty_callable`, but we need to update loads of test cases.